07-Extract-NDVI-From-Rasters-in-R.Rmd

```diff
@@ -161,104 +166,122 @@ names(avg_NDVI_HARV) <- "meanNDVI"
+approaches. 
 
-We'd like to produce a plot where Julian Days is on the X axis and NDVI is on the 
-Y axis. To do this, we'll need a column that is populated with Julian Day values
-for each NDVI value. We can do this using several approaches. For this lesson, 
-e will use `gsub` to replace the `X` and the `_HARV_ndvi_crop`.
+For this lesson, we will use `gsub` to replace the `X` and the `_HARV_ndvi_crop`.
 
 ```{r extract-julian-day }
-
-#note the use of | is equivalent to "or". this allows us to search for more than
-#one pattern in our text strings
+#note the use of the vertical bar character ( | ) is equivalent to "or". This
+# allows us to search for more than one pattern in our text strings.
 julianDays <- gsub(pattern = "X|_HARV_ndvi_crop", #the pattern to find 
             x = row.names(avg_NDVI_HARV), #the object containing the strings
             replacement = "") #what to replace each instance of the pattern with
 
+#alternate format
+#julianDays <- gsub("X|_HARV_ndvi_crop", "", row.names(avg_NDVI_HARV))
+
 #make sure output looks ok
 head(julianDays)
 
-#add julianDay values as a column in the dataframe
+#add julianDay values as a column in the data frame
 avg_NDVI_HARV$julianDay <- julianDays
 
 #what class is the new column
 class(avg_NDVI_HARV$julianDay)
-
-#convert julian days into a time class
-#avg_NDVI_HARV$julianDay <- as.POSIXlt(avg_NDVI_HARV$julianDay, format="%j")
-
-#the code above does weird things. for one i think jday is 0 based indexing
-#but also it seems to want a year and defaults to 2015. i am not sure how to force 
-#it to 2011 which is when the data were collected
-#being lazy and useing int for the time being.
-avg_NDVI_HARV$julianDay <- as.integer(avg_NDVI_HARV$julianDay)
-
-# Parse that character vector
-#strptime(date_info, "%Y %j")
 ```
 
-#Scale Data
-The last thing that we need to do it so scale our data. Valid NDVI values range 
-between 0-1. The metadata included a `ScaleFactor`. NDVI data are scaled by a 
-factor of 10,000. A scale factor is commonly used in larger datasets. Storing decimal
-places actually consumes more space and creates larger file sizes compared to 
-storing integer values.
-
-```{r scale-data }
 
-#scale data by 10,000
-avg_NDVI_HARV$meanNDVI <- avg_NDVI_HARV$meanNDVI / 10000
-#view output
-head(avg_NDVI_HARV)
+##Create Date Column from Julian Day
+Currently the Julian day is recognized as a character class value.  We'd prefer
+if it was recognized by `R` as one of the data-time classes so that `R` can
+interpret and format it as a date when plotting and in other functions. 
 
-```
+For more information on date-time classes, see the NEON Data Skills lesson 
+[Convert Date & Time Data from Character Class to Date-Time Class (POSIX) in R ]({{ site.baseurl}} /R/Time-Series-Convert-Date-Time-Class-POSIX/).  
 
-#Plot NDVI Using GGPLOT
+We know our data is from 2011, therefore the first day or origin for our Julian
```

Comment:
We need to walk the user through this as i think this will be a point of confusion.

##Creating a Date Class from Julian Days

Julian days, as used in these data, represents the number of days into a particular year, ranging from 1-365  (366 on a leap year). Julian days, by default does not define the actual year during which the data were collected. To convert a Julian Day value in `R` to a date class, we need to define the `origin` of the date (ie the starting date including the year).

<Feel free to edit the above> This is a good concept to introduce!